### ISSUE TITLE
Assert nr of variable names == dim or dim+1 is too restrictive for the depth coordinate system

### ISSUE THREAD

**Issue by anne-glerum** [geodynamics/aspect#2843]
When using the `depth` coordinate system in a `Function` for e.g. initial temperature, one intuitively sets the `Variable names` to _depth_. For time-independent functions, one would not add _time_, so only one variable name is set. This however triggers an AssertThrow in dealii that wants the number of variable names to be dim or dim+1. Of course the easy workaround is to set _time_ as a second variable name or to use the default name _x_, but the error message is inconsistent with the functionality of the depth coordinate system.

**Comment by gassmoeller** [comment]
Yes that is an issue about how we use the function with the depth coordinate system (differently from what deal.II expects). Adding time will only help in 2D though, and would be confusing, because *time* would really stand for *second spatial dimension that is never used*. A more consistent solution would be to add the following check before every call to `function.parse_parameters`, probably as a new utility function:

```
if (coordinate_system == Utilities::Coordinates::depth) 
  {
  std::string depth_variables = prm.get("Variable names");
  if (Utilities::split_string_list(depth_variables).size() == 1)
    {
      for (unsigned int i=1; i<dim; ++i)
        depth_variables += ", x" + std::to_string(i);

      prm.set("Variable names",depth_variables);
    }
```

This would append names like x1, x2 to the list of variables to fill it with sufficient entries. There is still a  problem in case someone writes a time-dependent function of depth. The user would provide 2 components, and the missing components would need to be added by the function between the two existing entries to make sure the time variable is last. Seems like a lot of work for an improvement in usability to me, but if you think it is important feel free to create a PR for it.

**Comment by anne-glerum** [comment]
I agree it's a lot of work to avoid triggering an Assert that is easily remedied from the input file, but after triggering it twice in a short time span I thought it was worth documenting :). Perhaps adding a sentence like "Note that upon changing the Variables names for the Depth coordinate system, dim or dim+1 names should be provided even though only the first variable is used." to the declaration of the Coordinate system could be useful?

**Comment by gassmoeller** [comment]
Yes that sounds like a reasonable solution.
